### PR TITLE
Add shared option to NnfSystemStorage

### PR DESCRIPTION
**Pull request by matthew-richerson**
The shared option determines whether to create a single allocation per Rabbit that is shared between all the computes on the Rabbit, or one allocation per compute node.

### CODE REVIEW

**Comment by bdevcich** [comment]
What's the reason for this change again?

File: api/v1alpha1/conversion.go [hunk 1]
```diff
@@ -486,8 +486,10 @@ func (src *NnfSystemStorage) ConvertTo(dstRaw conversion.Hub) error {
 	// Otherwise, you may comment out UnmarshalData() until it's needed.
 	if hasAnno {
 		dst.Spec.ExcludeDisabledRabbits = restored.Spec.ExcludeDisabledRabbits
+		dst.Spec.Shared = restored.Spec.Shared
 	} else {
 		dst.Spec.ExcludeDisabledRabbits = false
+		dst.Spec.Shared = true
```
Comment (bdevcich) [inline api/v1alpha1/conversion.go:492]: In alpha2/3, you have this set to false, but true here.

**Comment by matthew-richerson** [comment]
> What's the reason for this change again?

So that a XFS system storage can have a separate file system for each compute node.

File: api/v1alpha4/nnfsystemstorage_types.go [hunk 2]
```diff
@@ -83,6 +83,11 @@ type NnfSystemStorageSpec struct {
 	// +kubebuilder:default:=raw
 	Type string `json:"type,omitempty"`
 
+	// Shared will create one allocation per Rabbit rather than one allocation
+	// per compute node.
+	// +kubebuilder:default:=true
+	Shared bool `json:"shared"`
+
 	// StorageProfile is an object reference to the storage profile to use
```
Comment (bdevcich) [inline api/v1alpha4/nnfsystemstorage_types.go:91]: Where does the user set this? Gitops?

**Review by bdevcich — APPROVED** [review]

File: api/v1alpha4/nnfsystemstorage_types.go (hunk 2 above)
Comment (matthew-richerson) [reply api/v1alpha4/nnfsystemstorage_types.go:91]: Anywhere you can create a NnfSystemStorage resource. kubectl, gitops, etc.

**Review by roehrich-hpe — APPROVED** [review]

**Review by bdevcich — APPROVED** [review]

**Comment by matthew-richerson** [comment]
rebased on master to resolve conflicts

**Pull request merged**